### DISCUSSION TITLE
Buckling?

### DISCUSSION THREAD

**Discussion by jmhospers** [idaholab/moose#17328] [Q&A Modules: Solid mechanics] [answered]
Hello everyone,

I am looking at MOOSE for structural engineering applications, where we now use commercial applications. For many use cases, the high cost is not warranted and I want an alternative. MOOSE appears very capable for linear/plastic applications. However, I cannot find a capability for (linear bifurcation) buckling calculations? It must be possible, anyone here ever done this?

**Comment by hugary1995** [comment, marked as answer]
MOOSE currently doesn't have an arc-length solver. Contributions are always welcome.

**Reply by jmhospers** [reply]
I was hoping there was some obscure way to call the eigensolvers. I don't feel like my math/coding would be up to scrap at the moment.

**Reply by hugary1995** [reply]
Oh sorry! I misunderstood your question. MOOSE can definitely solve eigenvalue problems. Maybe use this page [https://mooseframework.inl.gov/source/executioners/NonlinearEigen.html](https://mooseframework.inl.gov/source/executioners/NonlinearEigen.html) as a starting point.

**Comment by aeslaughter** [comment]
@sapitts Do you have any information you could add about tensor mechanics?

**Reply by sapitts** [reply]
I don't have anything to add beyond what @hugary1995 has already posted about eigensolvers. @bwspenc would be a better bet for column buckling model recommendations
